lib/storage/http.js

```diff
@@ -22,15 +22,22 @@ const DEFAULT_CONTENT_TYPE = 'application/octet-stream';
 
 async function download(source, file) {
     try {
-        const contentType = mime.contentType(source.name)  || DEFAULT_CONTENT_TYPE;
+        const contentType = mime.contentType(source.name) || DEFAULT_CONTENT_TYPE;
         console.log(`downloading source: ${AssetComputeLogUtils.redactUrl(source.url)} into ${file}`);
+
+        const headers = {
+            'content-type': contentType
+        };
+        if(source.headers){
+            console.log('using additional custom headers');
+            Object.assign(headers, source.headers);
```

Comment:
wonder if we should only pass through only `Authorization` header instead of everything. If only custom authorization header is needed then console logs could be specific: `using custom authorization headers`

Reply:
In https://git.corp.adobe.com/mathern/drafts/pull/1, we decided to use the whole headers object.
In our case it's to passthrough Auth headers, but I wouldn't exclude other headers further down the line.